Add actions

Ughuuu opened the pull request:
Build mac, windows and linux automatically.
Also has cache.
A run takes about 1-2 minutes after cache is formed.

Ughuuu committed:
Add actions

With build, lint and upload
Runs on mac, linux and windows

add clang-format and tidy and update actions

run both format and tidy

add lint job/separate from build.

update builds for all platforms

checkout

use correct platform

add sconsflags

update targets

remove web build

try lint error

update dll's

run format and tidy

fix cache key

fix name issue

lint

fix trailing newline

Update linux_builds.yml

update

fix trailing endline

rburing commented on .github/actions/deps/action.yml:
```diff
@@ -0,0 +1,32 @@
+    - name: Setup clang-format
+      shell: bash
+      run: |
+        python -m pip install clang-format
```
Missing a trailing newline (sorry for nitpicking).

Ughuuu replied:
Fixed

rburing commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,60 @@
+          path: |
+            ${{ github.workspace }}/demo/bin/
+            ${{ github.workspace }}/demo/physics_server_box2d.gdextension
+          retention-days: 14
```
Missing a trailing newline (sorry for nitpicking).

Ughuuu replied:
Fixed

rburing reviewed (changes requested):
Thanks, this is looking great! See my comments below.

We could postpone the formatting if it's easier for you.

rburing commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,60 @@
+      - name: Lint
+        shell: sh
+        run:
+          clang-format
```
It would be nice to run this with `-i` on all code files and check the `git diff` as in Godot's [`clang-format.sh`](https://github.com/godotengine/godot/blob/950f4c3b41acef0403618dd6169be90530dea5da/misc/scripts/clang_format.sh).

I would prefer to make the empty diff a requirement, but we can do that later if you prefer.

Ughuuu replied:
Fixed

Ughuuu commented:
Sorry, just now read these. I since merged this branch into my fork though. I can update this branch though so you can merge it here.

Ughuuu commented:
Used same script from godot repo and linting now looks like this:
<img width="1047" alt="image" src="https://github.com/rburing/physics_server_box2d/assets/2369380/b3bb34aa-911e-4450-9d21-0e3361d35b1e">

Ughuuu commented:
Also updated the build to follow more the godot style and updated targets/caches:
<img width="836" alt="image" src="https://github.com/rburing/physics_server_box2d/assets/2369380/13be9899-b37c-4617-8682-63ae0ff603a0">

rburing commented on scripts/clang-format.sh:
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# This script runs clang-format and fixes copyright headers on all relevant files in the repo.
```
We don't have copyright headers (yet) and this script does nothing with them.

You might add a top comment that this is based on Godot's `clang-format.sh`.

rburing commented on scripts/clang-format.sh:
```diff
@@ -0,0 +1,36 @@
+
+if [ $# -eq 0 ]; then
+    # Loop through all code files tracked by Git.
+    files=$(git ls-files -- '*.c' '*.h' '*.cpp' '*.hpp' '*.cc' '*.hh' '*.cxx' '*.m' '*.mm' '*.inc' '*.java' '*.glsl' \
+                ':!:.git/*' ':!:thirdparty/*' ':!:*/thirdparty/*' ':!:platform/android/java/lib/src/com/google/*' \
+                ':!:*-so_wrap.*' ':!:tests/python_build/*')
```
These patterns and exclusions can be much simplified for this repository (and the same below).

Ughuuu replied:
simplified

rburing commented on scripts/clang-tidy.sh:
```diff
@@ -0,0 +1,33 @@
+set -uo pipefail
+
+# Loops through all code files tracked by Git.
+git ls-files -- '*.c' '*.h' '*.cpp' '*.hpp' '*.cc' '*.hh' '*.cxx' '*.m' '*.mm' '*.inc' '*.java' '*.glsl' \
+                ':!:.git/*' ':!:thirdparty/*' ':!:platform/android/java/lib/src/com/google/*' ':!:*-so_wrap.*' |
```
Same comments as for `clang-format.sh`.

Ughuuu replied:
same here, now only looks for .h and .cpp files

rburing reviewed (approved):
This is looking excellent now! I'm ready to hit the merge button, though addressing the final few comments would be ideal.

Ughuuu committed:
update clang format

Ughuuu commented:
Implemented feedback
